pkg/api/favorites.go

```diff
@@ -0,0 +1,76 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/common-fate/apikit/apio"
+	"github.com/common-fate/ddb"
+	"github.com/common-fate/granted-approvals/pkg/auth"
+	"github.com/common-fate/granted-approvals/pkg/storage"
+	"github.com/common-fate/granted-approvals/pkg/types"
+)
+
+// (GET /api/v1/favorites)
+func (a *API) UserListFavorites(w http.ResponseWriter, r *http.Request) {
```

Comment:
We need a delete endpoint here to get rid of a favorite a user creates accidentally